linode/rdns/framework_resource_schema.go

```diff
@@ -0,0 +1,47 @@
+package rdns
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+var frameworkResourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"address": schema.StringAttribute{
+			Description: "The public Linode IPv4 or IPv6 address to operate on.",
+			Required:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
+			},
+			Validators: []validator.String{
+				helper.NewIPStringValidator(),
+			},
+		},
+		"rdns": schema.StringAttribute{
+			Description: "The reverse DNS assigned to this address. For public IPv4 addresses, this will be set " +
+				"to a default value provided by Linode if not explicitly set.",
+			Required: true,
```

Comment:
This is very tricky because TF try to keep user configured values as the [SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth) but the Linode API will reset the value if this field is empty. So an empty field (null or an empty string) will be semantically equaled to the reset value (something like `45-80-141-70.ip.linodeusercontent.com`), and the custom type's semantic equivalence checking can't check it because it doesn't know the IP address in the [`StringSemanticEquals` function](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/custom-types#semantic-equality). The best way in my mind is to manually checking the semantic equivalence in the parse function of the model, I am not very sure.

Reply:
The previous implementation of this resource was only concerned with non-null (default) RDNS values since a Linode without RDNS should already use the default value. It might be worth adding this functionality though :+1:

Reply:
But I think we will still have to face the problem that an empty string semantically equals to the default rdns, right? This can happen when a user want to set the rdns to be the default value explicitly without manually concatenating the string.

Reply:
Ah, I miss the validator, so an empty string is not allowed?